### PR TITLE
Implement QSM's `CanDiff` via `tree-diff`

### PR DESCRIPTION
**Pull request by jasagredo**
# Description

Implement QSM's CanDiff via tree-diff

### CODE REVIEW

File: ouroboros-consensus/src/unstable-consensus-testlib/Test/Util/ToExpr.hs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+-- | This module implements QSM's @CanDiff@ typeclass using @tree-diff@'s
+-- @ToExpr@.
+module Test.Util.ToExpr () where
+
+import           Data.TreeDiff as T
+import qualified Test.StateMachine as QSM
+import           Test.StateMachine.Diffing (CanDiff (..))
+import qualified Test.StateMachine.Types.References as QSM
+
+instance ToExpr x => CanDiff x where
```
Comment (dnadales) [inline ouroboros-consensus/src/unstable-consensus-testlib/Test/Util/ToExpr.hs:16]: Would it make sense to add more comments on this typeclass, and perhaps explain how it is related to `ToExpr`?

Comment (jasagredo) [reply ouroboros-consensus/src/unstable-consensus-testlib/Test/Util/ToExpr.hs:16]: Well, this is the implementation. The comments live in https://github.com/stevana/quickcheck-state-machine/blob/00db0b4ebcf4faef434b98b0caaeeba0f17002b9/src/Test/StateMachine/Diffing.hs

Comment (jasagredo) [reply ouroboros-consensus/src/unstable-consensus-testlib/Test/Util/ToExpr.hs:16]: I think the module header should point the reader in the right direction to visit that other file
```
-- | This module implements QSM's @CanDiff@ typeclass using @tree-diff@'s
-- @ToExpr@.
```

Comment (dnadales) [reply ouroboros-consensus/src/unstable-consensus-testlib/Test/Util/ToExpr.hs:16]: Good point. I missed that.

**Review by jorisdral — APPROVED** [review]

**Pull request merged**